test/test_losses.py

```diff
@@ -9,6 +9,75 @@
 from torch.testing import assert_allclose
 
 
+class TestFocalLossWithLogits:
+    def test_smoke_none(self, device, dtype):
+        num_classes = 1
+        logits = torch.rand(2, num_classes, 3, 2, dtype=dtype, device=device)
+        labels = torch.rand(2, 3, 2, dtype=dtype, device=device)
+
+        assert kornia.losses.binary_focal_loss_with_logits(
+            logits, labels, alpha=0.5, gamma=2.0, reduction="none"
+        ).shape == (2, 3, 2)
+
+    def test_smoke_sum(self, device, dtype):
+        num_classes = 1
+        logits = torch.rand(2, num_classes, 3, 2, dtype=dtype, device=device)
+        labels = torch.rand(2, 3, 2, dtype=dtype, device=device)
+
+        assert (
+            kornia.losses.binary_focal_loss_with_logits(
+                logits, labels, alpha=0.5, gamma=2.0, reduction="sum"
+            ).shape == ()
+        )
+
+    def test_smoke_mean(self, device, dtype):
+        num_classes = 1
+        logits = torch.rand(2, num_classes, 3, 2, dtype=dtype, device=device)
+        labels = torch.rand(2, 3, 2, dtype=dtype, device=device)
+
+        assert (
+            kornia.losses.binary_focal_loss_with_logits(
+                logits, labels, alpha=0.5, gamma=2.0, reduction="mean"
+            ).shape == ()
+        )
+
+    def test_smoke_mean_flat(self, device, dtype):
+        num_classes = 1
+        logits = torch.rand(2, num_classes, 3, 2, dtype=dtype, device=device)
+        labels = torch.rand(2, 3, 2, dtype=dtype, device=device)
+
+        assert (
+            kornia.losses.binary_focal_loss_with_logits(
+                logits, labels, alpha=0.5, gamma=2.0, reduction="mean"
+            ).shape == ()
+        )
+
+    def test_jit(self, device, dtype):
+        num_classes = 1
+        logits = torch.rand(2, num_classes, 3, 2, dtype=dtype, device=device)
+        labels = torch.rand(2, 3, 2, dtype=dtype, device=device)
+
+        op = kornia.losses.binary_focal_loss_with_logits
+        op_script = torch.jit.script(op)
+        actual = op_script(logits, labels, alpha=0.5, gamma=2.0, reduction="none")
+        expected = op(logits, labels, alpha=0.5, gamma=2.0, reduction="none")
+        assert_allclose(actual, expected)
+
+    def test_gradcheck(self, device):
+        num_classes = 1
+        alpha, gamma = 0.5, 2.0  # for focal loss with logits
+        logits = torch.rand(2, num_classes, 3, 2).to(device)
+        labels = torch.rand(2, 1, 3, 2) * num_classes
+        labels = labels.to(device).long()
+
+        logits = utils.tensor_to_gradcheck_var(logits)  # to var
+        assert gradcheck(
+            kornia.losses.binary_focal_loss_with_logits,
+            (logits, labels, alpha, gamma),
+            raise_exception=True,
+        )
```

Comment:
add a small test comparing the results from the functional and the module

Reply:
Something like this? 

    def test_same_output(self, device, dtype):
        num_classes = 1
        logits = torch.rand(2, num_classes, 3, 2, dtype=dtype, device=device)
        labels = torch.rand(2, 3, 2, dtype=dtype, device=device)

        kwargs = {"alpha": 0.25, "gamma": 2.0, "reduction": 'mean'}

        assert kornia.losses.binary_focal_loss_with_logits(logits, labels, **kwargs) == \
               kornia.losses.BinaryFocalLossWithLogits(**kwargs)(logits, labels)